### PR TITLE
Add ability to remove files from staged file upload submission

### PR DESCRIPTION
**Pull request by balintbartok**
refs: MBL-14835
affects: Student
release note: Add ability to remove files from staged file upload submission

test plan:
* upload files for a submission
* gray X should be visible in each file's row
* click X to remove those files
* after confirmation files should be removed

### CODE REVIEW

File: Core/Core/UIViews/EmptyView.swift [hunk 1]
```diff
@@ -64,7 +64,7 @@ open class EmptyView: UIView {
         set { imageView?.image = newValue }
     }
 
-    public required init() {
+    public override init(frame: CGRect) {
```
Comment (balintbartok) [inline Core/Core/UIViews/EmptyView.swift:67]: IBDesignablesAgent was crashing on storyboard views constantly, with the error:
"Fatal error: Use of unimplemented initializer 'init(frame:)' for class 'Core.EmptyView'", this is a fix for that

**Comment by inst-danger** [comment]
<details><summary>Student.ipa</summary>
  <a href="https://app.bitrise.io/artifact/52848840/p/cb1d9bc7d594f3048209a357729d52cf">
    
  </a>
</details>
<details><summary>Teacher.ipa</summary>
  <a href="https://app.bitrise.io/artifact/52848848/p/fa70f448c8266c64117aa07e5c785ded">
    
  </a>
</details>
<details><summary>Parent.ipa</summary>
  <a href="https://app.bitrise.io/artifact/52848803/p/e3ce8c236e505a2ef1b91e7b0b7a7ea7">
    
  </a>
</details>

**Comment by inst-danger** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Fails</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:no_entry_sign:</td>
      <td>

  Build failed, skipping coverage check
  </td>
    </tr>
  </tbody>
</table>



<details><summary>:x: Swift lint</summary>


    ❌ /Users/vagrant/git/Core/Core/Files/FilePicker/FilePickerViewController.swift:369:1: Line Length Violation: Line should be 200 characters or less: currently 252 characters (line_length)
    [student]
    [teacher]
</details>
<details><summary>:x: XCTest failed: CoreTests/PageViewEventRequestManagerTests/testRetrievePandataEndpointInfo</summary>


    Assertion Failure at PageViewEventRequestManagerTests.swift:68: Asynchronous wait failed: Exceeded timeout of 5 seconds, with unfulfilled expectations: "events added".
    Assertion Failure at PageViewEventRequestManagerTests.swift:76: XCTAssertEqual failed: ("0") is not equal to ("2")
    Assertion Failure at PageViewEventRequestManagerTests.swift:87: XCTAssertEqual failed: ("Optional(["start", "end"])") is not equal to ("nil")
</details>


[MBL-14835](https://instructure.atlassian.net/browse/MBL-14835)

#### Affected Apps: Student

#### Release Note: 
Add ability to remove files from staged file upload submission
<p align="right">
  Generated by :no_entry_sign: <a href="http://github.com/danger/danger-js/">dangerJS</a>
</p>

File: Core/Core/Files/FilePicker/FilePickerViewController.swift [hunk 2]
```diff
@@ -362,3 +363,16 @@ extension FilePickerViewController: VNDocumentCameraViewControllerDelegate {
         }
     }
 }
+
+extension FilePickerViewController: FilePickerCellDelegate {
+    func removeFile(_ file: File) {
+        let title = NSLocalizedString("Remove File", bundle: .core, comment: "")
+        let message = NSLocalizedString("Are you sure you want to remove this file from the submission?", bundle: .core, comment: "")
```
Comment (vanwagonet) [inline Core/Core/Files/FilePicker/FilePickerViewController.swift:370]: I think this File Picker gets used for more than just assignment submissions, so we may want to remove the submission-specific text.

@armstrongnate Would `"Are you sure you want to remove this file?"` be ok?

**Review by vanwagonet — CHANGES_REQUESTED** [review]
The code looks great, just a nitpick about the text.

File: Core/Core/Files/FilePicker/FilePickerViewController.swift (hunk 2 above)
Comment (armstrongnate) [reply Core/Core/Files/FilePicker/FilePickerViewController.swift:370]: Yes, you're right. That would be better.

**Review by vanwagonet — APPROVED** [review]
QA+1

**Pull request merged**